lib/sna/auth/email.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Sna.Auth.Mail do
+
+  @moduledoc """
+
+  Sna.Auth.Mail will store email and token:
+
+  1. when an user comes and want to use email authentication, an email
+     is firstly checked.
+
+  2. if the mail is valid, a token is generated and the email is
+     stored temporarily in the ETS cache. A timer process is started
+     for some minutes.
+
+  3. if the token is valid (checked by the user), the mail is stored
+     in the underlaying database else, after some minutes, the mail
+     and token are rejected.
```

Comment:
Why do you need to store the e-mail in database ?

With JWT, the fact that the e-mail has been verified is stored in the token itself. if we don't need to store something else, I don't think we need the database just yet.

Reply:
What about the user state? If the session end (or the browser session is removed, or user want to use another computer)? We need to store those information somewhere. Not only on one session.